### DISCUSSION TITLE
v2.63.0

### DISCUSSION THREAD

**Discussion by yufeih** [dotnet/docfx#8527] [Announcement]
## New Modern Template

This release features a brand new `modern` site template based on bootstrap 5 with a lot of visual and usability improvements:

* Supports both dark and light themes
* Match .NET API reference docs with <https://learn.microsoft.com>
* Usability improvements such as code copy, clickable images, external link icons, better API text breaks, link to GitHub, etc.

The docfx official site has also been updated to use the new template. To use the modern template in your website, set the template to `["default", "modern"]`.

### 💥 Breaking Changes
* BREAKING CHANGE: drop support of liquid template by @yufeih in https://github.com/dotnet/docfx/pull/8437
### 🎉 New Features
* feat: new modern template @yufeih in https://github.com/dotnet/docfx/pull/8470 https://github.com/dotnet/docfx/pull/8440 https://github.com/dotnet/docfx/pull/8498 https://github.com/dotnet/docfx/pull/8515 https://github.com/dotnet/docfx/pull/8516 https://github.com/dotnet/docfx/pull/8524
* feat: add dark theme to modern template by @yufeih in https://github.com/dotnet/docfx/pull/8499 https://github.com/dotnet/docfx/pull/8512
* feat: adjust .NET API display for modern template by @yufeih in https://github.com/dotnet/docfx/pull/8525
* feat: show `_appName` after logo by @yufeih in https://github.com/dotnet/docfx/pull/8438
* feat: add code copy by @yufeih in https://github.com/dotnet/docfx/pull/8517
* feat: clickable image by @yufeih in https://github.com/dotnet/docfx/pull/8518
* feat: update docfx to use modern template by @yufeih in https://github.com/dotnet/docfx/pull/8513
* feat: add <wbr> tags to word boundaries by @yufeih in https://github.com/dotnet/docfx/pull/8521
* feat: add icon to external links by @yufeih in https://github.com/dotnet/docfx/pull/8523
* feat: show TOC node as expanded using the `expanded` property by @yufei in https://github.com/dotnet/docfx/pull/8498
* feat: Group explicit interface implementations into their own section by @mikernet in https://github.com/dotnet/docfx/pull/8445

### 🐞 Bug Fixes
* fix: Newtonsoft.Json.Schema -> JsonSchema.Net by @yufeih in https://github.com/dotnet/docfx/pull/8418
* fix: incorrect type name for unresolved generic types by @yufeih in https://github.com/dotnet/docfx/pull/8491
* fix: mref properties inherit parent items by @yufeih in https://github.com/dotnet/docfx/pull/8492
* fix: custom logo does not override logo from template by @yufeih in https://github.com/dotnet/docfx/pull/8510 https://github.com/dotnet/docfx/pull/8522

### 🔧 Engineering
* build: setup ts/scss/watch by @yufeih in https://github.com/dotnet/docfx/pull/8421 https://github.com/dotnet/docfx/pull/8423
* test: add visual snapshot testing by @yufeih in https://github.com/dotnet/docfx/pull/8422 https://github.com/dotnet/docfx/pull/8481
* build: upgrade to YamlDotNet 13.0.1 by @lahma in https://github.com/dotnet/docfx/pull/8442

### 📄 Documentation
* Add minimal template by @sotanakamura in https://github.com/dotnet/docfx/pull/8447
* Improvement Document Guideline markdown with image by @chuongmep in https://github.com/dotnet/docfx/pull/8485

## New Contributors
* @sotanakamura made their first contribution in https://github.com/dotnet/docfx/pull/8447
* @chuongmep made their first contribution in https://github.com/dotnet/docfx/pull/8485

**Full Changelog**: https://github.com/dotnet/docfx/compare/v2.62.2...v2.63.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dotnet/docfx/releases/tag/v2.63.0'>v2.63.0</a>.</em>

**Comment by VaclavElias** [comment]
This is so impressive!!! Thank you @yufeih. I will be migrating soon my community docs (https://doc.stride3d.net/latest/en/) to this new version and because I have been using also Bootstrap 5, I can reuse many things. 

![image](https://user-images.githubusercontent.com/4528464/226760512-0ae74b7a-115f-463b-8b7f-09f4f6a7d0c1.png)

![image](https://user-images.githubusercontent.com/4528464/226760862-166aa209-04e7-492c-b61f-62b7e74b4ea4.png)

**Comment by VaclavElias** [comment]
@yufeih I see that the **modern** template has some Bootstrap elements removed like `.row` and `col-`. Is that intended?

Also, do you plan to add to the **modern** template also Filter Page?

![image](https://user-images.githubusercontent.com/4528464/230740000-b91801bb-ebb1-4ee4-9141-8acba6571fb0.png)

I have finally started playing with the new docfx! 🙂

**Reply by VaclavElias** [reply]
I guess, I can use this `@import url("bootstrap-grid.min.css");` in my `main.css`.

**Reply by yufeih** [reply]
>  ... has some Bootstrap elements removed like .row and col-. Is that intended?

It seems I did set `$enable-grid-classes: false;` to clean up usage of `.row` and `.col` in favor of flex boxes and CSS grids. Do find find these bootstrap grid helpers useful? If so we can re-enable them.

> Also, do you plan to add to the modern template also Filter Page?

Yes!

**Reply by VaclavElias** [reply]
Thanks for removing `$enable-grid-classes: false;`. I don't have to hack the css to bring that part back. 

Also, my hack caused the issue with the logo. Now it is ok :). 

The usage of `.row` and `.col` doesn't bloat that css file that much.

**Comment by VaclavElias** [comment]
Also, I found a glitch, in the **modern** template, I can't figure out why the search box is under the scroll bar, the `container-xxl` should give it the correct padding. In Bootstrap examples, the scroll bar is next to the form elements..

![image](https://user-images.githubusercontent.com/4528464/230742899-c212d4ce-ea7d-403d-b9d2-d58810111954.png)

**Reply by yufeih** [reply]
Right, same as been reported in #8604

**Comment by VaclavElias** [comment]
Other glitches with the **modern** template, when the page is in mobile version..

1. Logo should be on the top, our "Stride" logo is longer so it would push the content even more
2. There are three dots
3. Search misalign
4. And again the right padding is missing across the whole page

![image](https://user-images.githubusercontent.com/4528464/230744507-7633d516-afd2-4347-bcce-44f1486f6895.png)

**Reply by yufeih** [reply]
It doesn't seem to appear in https://dotnet.github.io/docfx/ , is it because of some custom CSS?

**Reply by VaclavElias** [reply]
Yes, it was. It is now fixed.
